app/models/concerns/master_file_behavior.rb

```diff
@@ -116,7 +116,11 @@ def display_title
   end
 
   def embed_title
-    [media_object.title, display_title].compact.join(" - ")
+    [media_object.title, structure_title].compact.join(" - ")
+  end
+
+  def display_title
```

Comment:
For the least change in behavior how about making this method `structure_title` instead and using it in `stream_details`?

Reply:
I'm fine with doing that if you think it's best. I switched around the names for clarity, since this method is what is actually getting displayed but it also makes sense to change less code so I can go either way on it.

Reply:
I'm just afraid of unintended side effects of changing behavior of public methods, but clean up for clarity is important.  So I could also go either way too.